spec/models/enterprise_token_spec.rb

```diff
@@ -334,6 +342,18 @@
         expect(described_class.active_tokens).to be_empty
       end
     end
+
+    context "when the user messed with the database to 'extend' their validity" do
+      let!(:expired_token) { create_enterprise_token("an_active_token", expires_at: 1.day.ago) }
+
+      before do
+        expired_token.update(valid_until: 1.year.from_now)
+      end
+
+      it "returns an empty array" do
+        expect(described_class.active_tokens).to be_empty
+      end
+    end
```

Comment:
This test does not work: if the `.select { it.active? && !it.invalid_domain? }` part of `.set_active_tokens` is removed, it still passes. That's because the validity dates are extracted and saved in a `before_save` hook.

To make this test effective, `expired_token.update(valid_until: 1.year.from_now)` should be replaced by `expired_token.update_column(:valid_until, 1.year.from_now)` to avoid triggering the `before_save` hook, then the test will fail when the `.select { it.active? && !it.invalid_domain? }` part of `.set_active_tokens` is removed.

Reply:
true